docs/concepts/juttle_tutorial.md

```diff
@@ -2,27 +2,532 @@
+
+We will also try enriching the data with metadata from a CSV [file](../examples/datasets/github_metadata.csv), but not right away.
+
+## Exploring the data
+
+First, let's look at a few events using the `read file` source. We already used `head` and `tail` processors to limit the data we output, now we will try filtering by timestamp. Juttle defaults to looking for the timestamp in the field `time` unless `-timeField <fieldname>` option is used.
+
+XXX/dmehra this is currently unsupported:
+
+```
+read file -file '/tmp/data.json' -from :2015-07-01: -to :2015-07-02:
+```
+
+what works now is:
+
+```juttle
+read file -file '/tmp/data.json'
+| filter time >= :2015-07-01:
+```
+
+After running this program, we have our data (well, some of it) displayed in
+a table. Notice that each GitHub event has a timestamp, an event type, a repository name,
+and the login of the "actor" whose action triggered an event.
+
+## How many points? (Our first reduce)
+
+Let's answer a few basic questions about the data we are looking at. As a first
+step, we might want to know how many data points there are; the size of the data often informs how one would wish to analyze it. To do this, we'll use the 
+[`reduce`](../processors/reduce.md) processor that enables aggregations and grouping in Juttle.
+
+Let's look at aggregations first, and start by using the [`count()`](../reducers/count.md) reducer (aggregation functions are called "reducers" in Juttle), which simply counts the number of points that flow through it.
+
+```
+read file -file '/tmp/data.json'
+| reduce total_points = count()
+```
+
+You will have noticed that `reduce` trims off the rest of the data, leaving only the newly computed fields (`total_points` above). If you still have need for the original data for further processing, that can be done by splitting the flowgraph into `reduce` on one branch, and other logic on another branch; that is outside the scope of this tutorial. 
+
+Juttle comes with a number of built-in reducers to compute statistics such as
+counts, cardinality, averages, and derivatives. If you need a computation that isn't provided, you can define your own reducer.
+
+## Which fields? Which values? (More reduce magic)
+
+When working with a new dataset, one commonly wants to know how big the data is, and how it's structured in terms of fields and values. We know the total number of points in our data set, and we have an idea of what fields are in the points from doing `read ... | head` to look at a few points. However, those points may not have represented full variety of the data set; there may be other points in the middle of the dataset with additional fields that we didn't catch with `head` and `tail`.
+
+There is a Juttle trick to see which fields are present in the data; the processing will not be optimized, so do this only over small data sets (by specifying a narrow time range, for example).
```

Comment:
Could use help with the definition: would it be "data set that can fit in memory of a Node.js process"? what's the cap - 1GB? and how much of that can be used up by the data?